### PR TITLE
Add support for EML 2.0.0 and up

### PR DESCRIPTION
**Pull request by ThomasThelen**
This PR adds support for triplifying EML documents version 2.0.0 and up, fixing issue #63. Because of the backwards compatibilities of EML, new processing logic wasn't needed. Instead, new classes for each processor were set up and added to the [format map](https://github.com/DataONEorg/slinky/blob/715bd5210b86728cf29d0d06460beb80519d0fc0/d1lod/d1lod/client.py#L23). The unit tests make up the bulk of this PR.

### Changes

1. The base EMLProcessor class is the 2.0.0 processor
2. Added EML 2.0.1 and 2.1.0 processors, each use the base processor
3. Unit tests for each of the processors
4. EML URI change [here](https://github.com/DataONEorg/slinky/blob/715bd5210b86728cf29d0d06460beb80519d0fc0/d1lod/d1lod/client.py#L28)

### Testing
To test, make sure that the unit tests are passing. This is deployed on the dev environment, so you can optionally query it to look into the data further.

### CODE REVIEW

File: d1lod/d1lod/client.py [hunk 1]
```diff
@@ -18,8 +21,11 @@
 
 
 FORMAT_MAP = {
+    "eml://ecoinformatics.org/eml-2.0.0": EMLProcessor,
+    "eml://ecoinformatics.org/eml-2.0.1": EML201Processor,
+    "eml://ecoinformatics.org/eml-2.1.0": EML210Processor,
     "eml://ecoinformatics.org/eml-2.1.1": EML211Processor,
-    "https://eml.ecoinformatics.org/eml-2.2.0": EML220Processor,
+    "eml://ecoinformatics.org/eml-2.2.0": EML220Processor,
```
Comment (amoeba) [inline d1lod/d1lod/client.py]: This should be `https://eml.ecoinformatics.org/eml-2.2.0`.

**Review by amoeba — CHANGES_REQUESTED** [review]
Looks pretty good and tests pass, thanks. Let's fix the EML 2.2.0 format ID thing and then I think this is good to go.

File: d1lod/d1lod/client.py (hunk 1 above)
Comment (ThomasThelen) [reply d1lod/d1lod/client.py]: Thanks for catching it; fixed in [40e8016](https://github.com/DataONEorg/slinky/pull/69/commits/40e80169644898bf9ab2c92e7b4fed8eadb6b14e)

**Comment by amoeba** [comment]
Thanks @ThomasThelen.

**Pull request merged**